File: EasyLearning-BackEnd/src/main/java/com/hutech/easylearning/entity/ShoppingCart.java
package com.hutech.easylearning.entity;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import java.math.BigDecimal;
import java.text.DecimalFormat;
import java.time.LocalDateTime;
import java.util.Set;

@Entity
@Table(name = "shopping_cart")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ShoppingCart {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name ="id", columnDefinition = "VARCHAR(36)")
    String id;

    @Column(name = "shopping_cart_total_price", precision = 10, scale = 3)
    BigDecimal totalPrice;

    @Column(name = "user_id")
    String userId;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", referencedColumnName = "id", insertable = false, updatable = false)
    User user;

    @OneToMany(mappedBy = "shoppingCart", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    Set<ShoppingCartItem> shoppingCartItems;

    @Column(name = "date_create")
    LocalDateTime dateCreate;

    @Column(name = "date_change")
    LocalDateTime dateChange;

    @Column(name = "changed_by")
    String changedBy;

    @Column(name = "is_deleted")
    boolean isDeleted;
}

File: EasyLearning-Portal/src/model/ShoppingCart.ts
interface ShoppingCart {
  id: string;
  totalPrice: number;
  shoppingCartItems: ShoppingCartItem[];
  userId: string;
  dateCreate?: any;
  dateChange?: any;
  changedBy?: any;
  deleted: boolean;
}
interface ShoppingCartItem {
  id: string;
  cartItemName: string;
  quantity?: any;
  imageUrl: string;
  cartItemPrice: number;
  shoppingCartId: string;
  courseId: string;
  dateCreate: string;
  dateChange: string;
  changedBy: string;
  deleted: boolean;
}
